### PR TITLE
Working character recognition

### PR DESCRIPTION
**Pull request by moritzschaefer**

### CODE REVIEW

File: src/character_recognition/Cannyseparation.ipynb [hunk 1]
```diff
@@ -1,461 +0,0 @@
-{
```
Comment (shansmann) [inline src/character_recognition/Cannyseparation.ipynb:1]: maybe put visualizations in notebook folder, add path to notebook

**Review by shansmann — APPROVED** [review]
messed it up

File: src/Cannyseparation.ipynb [hunk 2]
```diff
@@ -0,0 +1,443 @@
+{
```
Comment (shansmann) [inline src/Cannyseparation.ipynb]: move into notebook folder

File: src/character_recognition.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    '''
+    Calculates the texts in an img
+    :img: The original image
+    :text_probability: The text recognition probability image. pixels
+    should be probablities in range [0,1]
+    :return: A list of dictionaries, each containing position and text
+
+    '''
+    # TODO tweak param
+    THRESHOLD = 0.7
```
Comment (shansmann) [inline src/character_recognition.py]: move to config

File: src/character_recognition.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+from character_training import load_model
+import config
+
+
+def predict_wordfile(filename, model, dictionary):
+    '''
+    Load a file and predict the whole image as word
+    '''
+    raise ValueError('Not implemented...')
+
```
Comment (shansmann) [inline src/character_recognition.py:25]: i guess u can use text_prediction_img.npy

Comment (moritzschaefer) [reply src/character_recognition.py:25]: Issue is that i need to generate a probability map an stuff like that for character recognition to run..

File: src/character_recognition.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    try:
+        logging.info('Trying to load model')
+        model = load_model()
+    except FileNotFoundError:  # noqa
+        logging.warn('Model not found, please run character training')
+        sys.exit(1)
+
+    filename = os.path.join(config.TEXT_PATH, '2/104.jpg')
+
+    get_predictions(filename, model, dictionary)
```
Comment (shansmann) [inline src/character_recognition.py:132]: will produce pretty neat results with the right input!  :)

**Pull request merged**